eventastic_postgres/src/transaction.rs

```diff
@@ -222,6 +256,21 @@ where
             .pickle()
             .context("Failed to pickle aggregate")
             .map_err(DbError::PicklingError)?;
+        let cipher = self
+            .encryption_provider
+            .encrypt(vec![aggregate])
+            .await
+            .context("Failed to encrypt snapshot")
+            .map_err(DbError::Encryption)?;
+        if cipher.len() != 1 {
+            return Err(DbError::Encryption(anyhow!(
+                "Encrypting snapshot returned wrong number of items"
+            )));
+        }
+        let aggregate = cipher
+            .into_iter()
+            .next()
+            .expect("Must have encrypted snapshot");
```

Comment:
I dislike these two checks being independent as we could easily remove the check and then start having panics. 

Can we do just one check and get the aggregate at the same time e.g

```suggestion
        let aggregate = cipher.into_iter().next().ok_or_else(|| {
            DbError::Encryption(anyhow!("Encrypting snapshot returned empty result"))
        })?;
```

Reply:
That change isn't quite the same, if the encryption provider returns more items then that won't catch it. So it's still two statements but it doesn't use `.expect` anymore.